studio/src/editors/version-panel.js

```diff
@@ -0,0 +1,775 @@
+                                          `
+                                        : nothing}
+                                </div>
+                            </div>
+                            <div class="version-meta">
+                                <div class="version-date">
+                                    <svg width="12" height="12" viewBox="0 0 24 24" fill="currentColor">
+                                        <path
+                                            d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zM7 10h5v5H7z"
+                                        />
+                                    </svg>
+                                    ${this.formatVersionDate(version.created)}
+                                </div>
+                                <div class="version-author">
+                                    <svg width="12" height="12" viewBox="0 0 24 24" fill="currentColor">
+                                        <path
+                                            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
+                                        />
+                                    </svg>
+                                    By ${version.createdBy || 'Unknown'}
+                                </div>
+                            </div>
+                            ${version.comment ? html`<div class="version-description">${version.comment}</div>` : nothing}
+                        </div>
+                    `;
+                })}
+            </div>
+        `;
+    }
+
+    get versionActions() {
+        return nothing;
+    }
+
+    render() {
+        return html`
+            <sp-action-button
+                label="Version History"
+                title="View version history"
+                ?disabled="${this.disabled || this.loading}"
+                @click="${this.togglePanel}"
+            >
+                ${this.versionHistoryIcon}
+                <sp-tooltip self-managed placement="bottom">Version History</sp-tooltip>
+            </sp-action-button>
+
+            ${this.showPanel
+                ? html`
+                      <div class="version-history-panel">
+                          <div class="version-history-header">
+                              <button class="back-button" @click="${this.closePanel}" title="Back to Editor">
+                                  ${this.backArrowIcon}
+                              </button>
+                              <h3 class="version-history-title">Version History</h3>
+                          </div>
+                          ${this.versionList}
+                      </div>
+                  `
+                : nothing}
+            ${this.showEditModal
+                ? html`
+                      <div class="edit-modal-backdrop" @click="${this.closeEditModal}">
```

Comment:
I know the UI says this is not implemented but I see a non spectrum dialog, we should avoid commiting custom UX code.

<img width="477" height="350" alt="image" src="https://github.com/user-attachments/assets/3419804e-b775-497e-9a5f-459f85229343" />